src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
+            zoneJson.SetNamedValue(L"height", json::value(height));
+            zonesJson.Append(zoneJson);
         }
-
-        return res;
+        infoJson.SetNamedValue(L"zones", zonesJson);
+        return infoJson;
     }
 
-    bool FancyZonesData::ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
+    std::optional<FancyZonesDataTypes::CanvasLayoutInfo> CanvasLayoutInfoJSON::FromJson(const json::JsonObject& infoJson)
     {
-        std::scoped_lock lock{ dataLock };
         try
         {
-            auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
+            FancyZonesDataTypes::CanvasLayoutInfo info;
+            info.lastWorkAreaWidth = static_cast<int>(infoJson.GetNamedNumber(L"ref-width"));
+            info.lastWorkAreaHeight = static_cast<int>(infoJson.GetNamedNumber(L"ref-height"));
 
-            for (uint32_t i = 0; i < appLastZones.Size(); ++i)
+            json::JsonArray zonesJson = infoJson.GetNamedArray(L"zones");
+            uint32_t size = zonesJson.Size();
+            info.zones.reserve(size);
+            for (uint32_t i = 0; i < size; ++i)
             {
-                json::JsonObject appLastZone = appLastZones.GetObjectAt(i);
-                if (auto appZoneHistory = AppZoneHistoryJSON::FromJson(appLastZone); appZoneHistory.has_value())
-                {
-                    appZoneHistoryMap[appZoneHistory->appPath] = std::move(appZoneHistory->data);
-                }
-                else
-                {
-                    return false;
-                }
+                json::JsonObject zoneJson = zonesJson.GetObjectAt(i);
+                const int x = static_cast<int>(zoneJson.GetNamedNumber(L"X"));
+                const int y = static_cast<int>(zoneJson.GetNamedNumber(L"Y"));
+                const int width = static_cast<int>(zoneJson.GetNamedNumber(L"width"));
+                const int height = static_cast<int>(zoneJson.GetNamedNumber(L"height"));
+                FancyZonesDataTypes::CanvasLayoutInfo::Rect zone{ x, y, width, height };
+                info.zones.push_back(zone);
             }
-
-            return true;
+            return info;
         }
         catch (const winrt::hresult_error&)
         {
-            return false;
+            return std::nullopt;
         }
     }
 
-    json::JsonArray FancyZonesData::SerializeAppZoneHistory() const
+    json::JsonObject GridLayoutInfoJSON::ToJson(const FancyZonesDataTypes::GridLayoutInfo& gridInfo)
     {
-        std::scoped_lock lock{ dataLock };
-        json::JsonArray appHistoryArray;
+        json::JsonObject infoJson;
+        infoJson.SetNamedValue(L"rows", json::value(gridInfo.m_rows));
+        infoJson.SetNamedValue(L"columns", json::value(gridInfo.m_columns));
+        infoJson.SetNamedValue(L"rows-percentage", NumVecToJsonArray(gridInfo.m_rowsPercents));
+        infoJson.SetNamedValue(L"columns-percentage", NumVecToJsonArray(gridInfo.m_columnsPercents));
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.